### PR TITLE
[WIP] add time series workbook from NCEAS training

### PR DESCRIPTION
**Pull request by dlebauer**
This is a first draft of tomorrow's time series workshop. Feedback appreciated. I will be working on it today.

### CODE REVIEW

File: 202112_timeseries/time-series-forecasting.Rmd [hunk 1]
```diff
@@ -0,0 +1,699 @@
...
+Now that we are working with our 'population' of model fits, we can use all manner of statistics and visualizations to better understand the simulated experiments. We can use `pairs`, `hist`, `quantile` and others. 
+
+For the classical 'power analysis' lets look at the percent of time the slope term is significant at 
+P <= 0.05. How often is the slope significant? How often is the intercept significant?
+
+This is rather straightforward with the 'stats' data.frame we created above:
+
+```{r}
+
+sum(intercept_ps > 0.05)/length(day_ps)
```
Comment (jessicaguo) [inline 202112_timeseries/time-series-forecasting.Rmd:160]: intercept_ps and day_ps don't exist at this point

Comment (jessicaguo) [reply 202112_timeseries/time-series-forecasting.Rmd:160]: `sum(stats$p_i > 0.05)/nrow(stats)`
`sum(stats$p_s > 0.05)/nrow(stats)`
Is this what was intended?

File: 202112_timeseries/time-series-forecasting.Rmd [hunk 2]
```diff
@@ -0,0 +1,699 @@
...
+```
+
+
+This plot shows the four panels. First is the raw data, the the next three panels show the trend, seasonal pattern, and the residuals. These can be used in further statistical analysis. For example, you may be interested in learning about other variables that explain patterns in the the residuals after the trend and seasonal patterns are removed. 
+
+Lets simulate the number of times the fish sees a twig float by, and see if that affects our time series:
+
+```{r}
+r <- rpois(n = length(dec$random), lambda = 4)
+tidy(lm(dec$random ~ r, data = dec_df))
```
Comment (jessicaguo) [inline 202112_timeseries/time-series-forecasting.Rmd:328]: Not sure how dec_df was produced

File: 202112_timeseries/time-series-forecasting.Rmd [hunk 3]
```diff
@@ -0,0 +1,699 @@
...
+**I: Integrated**
+**MA: Moving Average smoothing**
+
+- _p_ number of lags 
+- _d_ degree of differencing
+- _q_ size of moving average window
+
+The `forecast::auto.arima` function automatically fits these parameters. See https://otexts.com/fpp2/arima-r.html for an explanation of how this function works.
+
+**STL** Seasonal Seasonal Decomposition of Time Series by Loess. Extracts trend, seasonal, and locally smoothed moving average (described above).
```
Comment (jessicaguo) [inline 202112_timeseries/time-series-forecasting.Rmd:392]: Duplicate 'Seasonal'

File: 202112_timeseries/time-series-forecasting.Rmd [hunk 4]
```diff
@@ -0,0 +1,699 @@
...
+The `forecast::auto.arima` function automatically fits these parameters. See https://otexts.com/fpp2/arima-r.html for an explanation of how this function works.
+
+**STL** Seasonal Seasonal Decomposition of Time Series by Loess. Extracts trend, seasonal, and locally smoothed moving average (described above). 
+
+**ETS** Exponential smoothing state space model. These forecast based on previous observations, but the importance of the prior observations are weighted using exponential smoothing such that more recent observations have more weight. See https://otexts.com/fpp2/expsmooth.html
+
+
+### Tools for forecasting
+
+As we discussed last week, there are some simple or 'naiive' ways to forecast - using the historical means from a seasonal trend, or using the overall mean. These approaches are useful, sometimes because they perform well and other times because they provide a reasonable null hypothesis. Some use the same methods that we used to interpolate.
```
Comment (jessicaguo) [inline 202112_timeseries/time-series-forecasting.Rmd:399]: 'naive'

File: 202112_timeseries/time-series-forecasting.Rmd [hunk 5]
```diff
@@ -0,0 +1,699 @@
...
+
+The `forecast` package has a lot of handy functions for the analysis of time series data, and the fitting of complex models including components described above. Here are some that may be useful:
+
+There are also functions `ets`, `stlm`, `ma`, `spline`, `meanf`, and `snaive` for ETS, STL, Moving Average, Spline, and Naive forecasts. 
+
+The `tslm` function uses `lm` and adds the ability to automatically specify trend and season componets as predictors (even if they aren't in the data frame). The funciton fits a linear regression to trend, season, and other factors. This has special terms 'trend' and 'season' that don't actually have to be in the data frame - they are automatically fit and added. The result will include one estimate for the trend, and another for each time step (in this case, an intercept for the first month and a parameter for each subsequent month):
```
Comment (jessicaguo) [inline 202112_timeseries/time-series-forecasting.Rmd:405]: 'components', 'function'

File: 202112_timeseries/time-series-forecasting.Rmd [hunk 6]
```diff
@@ -0,0 +1,699 @@
...
+
+Indeed, whatever is used to actually measure - often boils down to changes in an electric current that capture a property of interest. If we are measuring temperature with a mercury thermometer, there is an 'observation' model (scale on the thermometer) that converts the observation of the height of mercury into the latent variable of air temperature.
+
+A common application of state space modeling is to the study of dynamical systems. Often such a model is a set of equations that describe how a system evolves in time. 
+
+As a simple example, consider a model of a pendulum (see [Wikipedia](https://en.wikipedia.org/wiki/Pendulum_(mechanics)). This model has terms for the force of gravity and the length of the pendulum. The observatino might be the location of the pendulum at multiple points in time. It is then possible to estimate the force of gravity, the torque, velocity, and potential and kinetic energy at any point in time by fitting the observations of position to the dynamical model of pendulum dynamics.
```
Comment (jessicaguo) [inline 202112_timeseries/time-series-forecasting.Rmd:433]: 'observation'

File: 202112_timeseries/time-series-forecasting.Rmd [hunk 7]
```diff
@@ -0,0 +1,699 @@
...
+Lets look at some of the lags in the time series of temperature. The lags we look at will be 
+1 day, 1 week, 1 month, 1 quarter, 1/2 year and 1 year.
+
+```{r}
+lag.plot(tmean.ts, set.lags = c(1, 7, 30, 90, 180, 360))
+```
+
+Which of these lags are positively correlated? 
+
+<detalis>The day, week, and month are all positively correlated.</details>
```
Comment (jessicaguo) [inline 202112_timeseries/time-series-forecasting.Rmd:550]: 'details'

**Review by jessicaguo — COMMENTED** [review]
Nice overview with exercises!

File: 202112_timeseries/time-series-forecasting.Rmd (hunk 2 above)
Comment (dlebauer) [reply 202112_timeseries/time-series-forecasting.Rmd:328]: not sure; removed

**Comment by dlebauer** [comment]
Thanks for your feedback @jessicaguo!

**Pull request merged**